app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt

```diff
@@ -328,6 +348,7 @@ class LocationSensorManager : LocationSensorManagerBase() {
     }
 
     private fun stopHighAccuracyService() {
+        isBackgroundLocationSetup = false
```

Comment:
Won't this create problems when this function is called from [here](https://github.com/home-assistant/android/blob/90a9a27e5e371445f75481e7a93609528357d3ff/app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt#L271)? `requestLocationUpdates()`, which is called in the next line, doesn't set it back to `true`.

Reply:
it should not because it will only hit this part of the condition, thats why I changed it : 

https://github.com/home-assistant/android/blob/90a9a27e5e371445f75481e7a93609528357d3ff/app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt#L249-L255

Reply:
How would it end up in `setupBackgroundLocation` again? We cannot rely on when the SensorWorker starts.

Reply:
Because after this condition, the next line hits `setupBackgroundLocation` as part of the same method `setupLocationTracking`

https://github.com/home-assistant/android/blob/fc7c6b8316d4ecc30fd640e3df5f5dc9b87ccdc1/app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt#L235-LL248

Immediately followed by: https://github.com/home-assistant/android/blob/fc7c6b8316d4ecc30fd640e3df5f5dc9b87ccdc1/app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt#L250

Reply:
I linked to `stopHighAccuracyService` from deeper inside `setupBackgroundLocation` when switching high accuracy mode from on -> off, so not in `setupLocationTracking` and after the "it will only hit this part of the condition" lines you linked to :)

Reply:
ok I see what you mean, will move the logic to be inside the condition. Thanks for further clarifying your first comment 🤦

Reply:
No problem and thanks for your patience, the code is complex and I easily could've missed something

Reply:
Yea these changes are not easy, and things are easy to miss as a result.